src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,21 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Machine> {
```

Comment:
The `TruckProducer` should be parameterized with `Truck` instead of `Machine` to prevent the creation of producers that do not use `Machine` or its subtypes.